### ISSUE TITLE
[feature] Add Support for <iframe> components

### ISSUE THREAD

**Issue by seflless** [tldraw/tldraw#396]
It would be hugely valuable to be able to drop iframes of any webpage into a canvas.

**Comment by proful** [comment]
What about the initial size? It is going to be fixed for example 500x500?

**Comment by FarazzShaikh** [comment]
@proful @steveruizok this should be fairly straightforward and very cool to have as well. I can knock this one out in a couple days. Can you assign to me?

**Comment by steveruizok** [comment]
Yeah, I will. My biggest hesitation here is on the UI: we may need some kind of address bar as part of the shape itself, perhaps only visible when the shape is selected. We'll also need to make these shapes "editable" when double clicked, and only take pointer events while editing, so that for example scrolling over an iframe shape doesn't capture the script within the frame. But take a shot at it and we'll work up from there!

**Comment by FarazzShaikh** [comment]
Yeah you’re right. As you suggest, let’s have an “focus” state enabled by double clicking which then passes on mouse events to the iframe instead of the canvas. Deactivated by “esc” key perhaps?

for the address bar, maybe that can be in the right-click context menu? As the double click action will be takes by the “focus” mode.

Also it’s going to be virtually impossible to sync up the internal state of an iFrame over multiplayer. What do you suggest we do? 

Do we make multiplayer iFrames independent? I guess even for Video, we are not syncing the playback across clients so I guess it’s a fair compromise?

**Comment by seflless** [comment]
@FarazzShaikh a lot of embeddable documents are now multiplayer themselves. So this composed well.

**Comment by proke03** [comment]
@steveruizok 
Looks like the assignee has stopped working.
